File: ros_tutorial_py/scripts/requester.py
#!/usr/bin/env python
'''
  Filename: requester.py
  Author: Ryan Shim
'''

import rospy
from ros_tutorial_py.srv import Ask, AskRequest

def client():
    rospy.wait_for_service('ask')
    try:
        ask_cli = rospy.ServiceProxy('ask', Ask)
        req = AskRequest()
        req.question = "ryan smart?"
        res = ask_cli(req)
        print "Requesting: %s"%req.question
        print "Response: %s"%res.result
    except rospy.ServiceException, e:
        print "Failed to call service ask: %s"%e

if __name__ == "__main__":
    client()

File: ros_tutorial_cpp/src/requester.cpp
/*
  Filename: requester.cpp
  Author: Ryan Shim
*/

#include <ros/ros.h>
#include "ros_tutorial_cpp/Ask.h"

int main(int argc, char **argv)
{
  ros::init(argc, argv, "requester");
  ros::NodeHandle nh;
  ros::ServiceClient ask_cli = nh.serviceClient<ros_tutorial_cpp::Ask>("ask");
  ros_tutorial_cpp::Ask ask;
  ask.request.question = "ryan smart?";
  if (ask_cli.call(ask))
  {
    ROS_INFO("Request: %s", ask.request.question.c_str());
    ROS_INFO("Response: %s", ask.response.result.c_str());
  }
  else
  {
    ROS_ERROR("Failed to call service ask");
    return 1;
  }
  return 0;
}
